### PR TITLE
[improve][doc] Java example explaining code without using lombok

### PR DESCRIPTION
**Pull request by AlvaroStream**
### Motivation

Following strictly the example can make people that doesn't use lombok get this error:

No serializer found for class com.mytest.Product and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS)

### Modifications

Increase the documentation on the code example.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [X] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

I tried executing the sh start.sh:

````

[SUCCESS] Docusaurus website is running at: http://localhost:3000/

✔ Client
  Compiled successfully in 27.76s

client (webpack 5.75.0) compiled with 1 warning
<w> [webpack.cache.PackFileCacheStrategy] Skipped not serializable cache item 'Compilation/modules|/Users/arodriguez/Code/git_pulsar_alvaro/site2/.preview/pulsar-site/site2/website-next/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[6].use[1]!/Users/arodriguez/Code/git_pulsar_alvaro/site2/.preview/pulsar-site/site2/website-next/node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[6].use[2]!/Users/arodriguez/Code/git_pulsar_alvaro/site2/.preview/pulsar-site/site2/website-next/src/css/custom.css': No serializer registered for Warning
<w> while serializing webpack/lib/cache/PackFileCacheStrategy.PackContentItems -> webpack/lib/NormalModule -> Array { 1 items } -> webpack/lib/ModuleWarning -> Warning

````

But I couldn't access the docs:

http://localhost:3000/docs/schema-understand/
http://localhost:3000/docs/2.10.x/schema-understand/

Was giving always `Page Not found`

### Matching PR in forked repository

PR in forked repository: https://github.com/AlvaroStream/pulsar

### CODE REVIEW

File: site2/docs/schema-understand.md [hunk 1]
```diff
@@ -122,6 +122,11 @@ Pulsar gets the schema definition from the predefined `struct` using an Avro lib
 1. Create the _User_ class to define the messages sent to Pulsar topics.
 
    ```java
+   # If you use Lombok
+   @Builder
+   @AllArgsConstructor
+   @NoArgsConstructor
```
Comment (codelipenghui) [inline site2/docs/schema-understand.md:128]: We can remove line 133 to 137 since lombok will generate the constructors.

Comment (AlvaroStream) [reply site2/docs/schema-understand.md]: Thanks I just changed it a i though it was.

Removed.

**Review by RobertIndie — APPROVED** [review]

**Comment by codelipenghui** [comment]
@AlvaroStream 

I noticed you have removed

```
   @Builder
    @AllArgsConstructor
    @NoArgsConstructor
```

From the last comment https://github.com/apache/pulsar/pull/19019#discussion_r1055114376. We don't need to add the constructors if we have the annotation. The Lombok will generate them. 

We should remove https://github.com/apache/pulsar/pull/19019/files#diff-fe3c06d8365d562427069af4376e699002169fdcf1047e836e771ce355756044R130-R134 if we have Lombok annotations.

**Comment by AlvaroStream** [comment]
@codelipenghui sorry, I missunderstood. Now is better?

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**